### PR TITLE
gofed client: initial import

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: client.py [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+import sys
+import urllib2
+import json
+import logging
+from datetime import datetime
+from dateutil.parser import parse as datetime_parse
+
+# Constants
+DEFAULT_DEPTH = 20
+GOFED_WEB_URL = "http://209.132.179.123/"
```
Comment (ingvagabund) [inline client.py]: DEFAULT_DEPTH AND GOFED_WEB_URL should go into /etc/gofed.conf.

See b994c0104bf07d1bc14ee930278bcc8203ce2161

**Comment by ingvagabund** [comment]
Thanks for the pull request, Frido. At the moment it looks like the server is down. Can you checkout where is the problem? Got some questions and nits:
- dateutil module is required. Why do we need it?
- could you create module.RESTClient module and move all get_\* functions there into RESTClient class? Which would be called from client.py file. I would like to have all top level scripts be as short as possible and call all methods from modules. At the same time RESTClient could be used by other modules, e.g. by 'gofed check-deps' command with --server or similar command to query the server instead of local databases.
- format_filter and normal_format_str functions could be moved into modules.Utils modules if it can be used by other modules. Or at least put them on the same level as the remaining functions in RESTClient class.
- some parser.add_option_group looks quiet scary :) Can you fix formatting?

**Comment by fridex** [comment]
> - dateutil module is required. Why do we need it?
>   It is suitable for parsing dates. A user can be forced to strictly use 
>   YYYY-MM-DD  format if you want to omit parsing feature. Passed date has 
>   to be checked then (some simple check at least). Your opinion?
> - could you create module.RESTClient module and move all get_*
>   functions there into RESTClient class? Which would be called from
>   client.py file. I would like to have all top level scripts be as
>   short as possible and call all methods from modules. At the same
>   time RESTClient could be used by other modules, e.g. by 'gofed
>   check-deps' command with --server or similar command to query the
>   server instead of local databases.
>   Sounds reasonable.
> - format_filter and normal_format_str functions could be moved into
>   modules.Utils modules if it can be used by other modules. Or at
>   least put them on the same level as the remaining functions in
>   RESTClient class.
>   OK
> - some parser.add_option_group looks quiet scary :) Can you fix
>   formatting?
>   Will check; we could use some formatter for sources. BTW with tabstop=4 
>   the source looks OK to me.

I will prepare modified patch with improvements.

**Comment by fridex** [comment]
Modified patch with improvements. Formatting needs to be fixed according to (future) coding style.

**Comment by fridex** [comment]
Implements #32

**Comment by ingvagabund** [comment]
Nice :)

http://209.132.179.123/rest/check-deps/cadvisor/6ee904d12955715f2c7286c252827a35300bc027

**Pull request merged**